spring-cloud-dataflow-docs/src/main/asciidoc/task-developer-guide.adoc

```diff
@@ -57,118 +57,233 @@ If you have debug mode turned on, you can get even more information, in a line s
+additional configuration registers the `TaskRepository` and the infrastructure for its
+use.
+
+In our demo, the `TaskRepository` uses an embedded H2 database to record the results
+of a task. This H2 embedded database is not a practical solution for a production environment, since
+the H2 DB goes away once the task ends. However, for a quick getting-started
+experience, we will use this in our example as well as echoing to the logs what is being updated
+in that repository. In the <<features-configuration>> section (later in this
+documentation), we cover how to customize the configuration of the pieces provided by
+Spring Cloud Task.
+
+When our sample application runs, Spring Boot launches our `TimestampCommandLineRunner`
+and outputs our timestamp message to standard out. The `TaskLifecycleListener`
+records the start of the task and the end of the task in the repository.
+
+[[getting-started-main-method]]
+==== The main method
+
+The main method serves as the entry point to any java application.  Our main method
+delegates to Spring Boot's `SpringApplication` class.  You can read more about it in the
+Spring Boot documentation.
+
+[[getting-started-clr]]
+==== The CommandLineRunner
+
+Spring includes many ways to bootstrap an application's logic. Spring Boot provides
+a convenient method of doing so in an organized manner through its `*Runner` interfaces
+(`CommandLineRunner` or `ApplicationRunner`). A well behaved task can bootstrap any
+logic by using one of these two runners.
+
+The lifecycle of a task is considered from before the `*Runner#run` methods are executed
+to once they are all complete. Spring Boot lets an application use multiple
+`*Runner` implementations, as does Spring Cloud Task.
+
+NOTE: Any processing bootstrapped from mechanisms other than a `CommandLineRunner` or
+`ApplicationRunner` (by using `InitializingBean#afterPropertiesSet` for example) is not
+ recorded by Spring Cloud Task.
+
+[[getting-started-running-the-example]]
+=== Running the Example
+
+At this point, our application should work.  Since this application is Spring Boot-based,
+we can run it from the command line by using `$ mvn spring-boot:run` from the root
+of our application, as shown (with its output) in the following example:
+
+[source]
+----
+$ mvn clean spring-boot:run
+....... . . .
+....... . . . (Maven log output here)
+....... . . .
+
+  .   ____          _            __ _ _
+ /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
+( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
+ \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
+  '  |____| .__|_| |_|_| |_\__, | / / / /
+ =========|_|==============|___/=/_/_/_/
+ :: Spring Boot ::        (v2.0.3.RELEASE)
+
+2018-07-25 11:11:16.695  INFO 5342 --- [           main] i.s.demo.timestamp.TimestampApplication  : Starting TimestampApplication on Glenns-MacBook-Pro-2.local with PID 5342 (/Users/glennrenfro/project/crap/timestamp/target/classes started by glennrenfro in /Users/glennrenfro/project/crap/timestamp)
```

Comment:
May not want "crap" in your path in the docs...

Reply:
Good catch :-D

Reply:
removed